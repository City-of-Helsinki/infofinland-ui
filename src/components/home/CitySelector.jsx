import Button from '@/components/Button'
import { useRouter } from 'next/router'
import Image from 'next/image'
// import image from '@/public/images/suomi-kartta.png'
import Block from '@/components/layout/Block'
const CitySelector = () => {
  const { push } = useRouter()
  const goToCities = () => push('/cities')
  return (
    <Block className="pt-8 border-t border-gray-darker">
      <div className="grid grid-cols-2 pb-8">
        <div className="foo">
          <h2 className="mb-4 text-h2">Cities</h2>
          <p className="mb-8">
            Purus non blandit sem consectetur et dignissim arcu pellentesque
            morbi neque.
          </p>

          <Button onClick={goToCities}>Go to Cities</Button>
        </div>
        <div className="flex justify-center">
          <div className="">
            <Image
              src="/images/suomi-kartta.png"
              alt=""
              height="381"
              width="219"
              layout="intrinsic"
            />
          </div>
        </div>
      </div>
    </Block>
  )
}

export default CitySelector
