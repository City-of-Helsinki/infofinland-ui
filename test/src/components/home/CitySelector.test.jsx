/**
 * @jest-environment jsdom
 */
/* eslint-disable import/no-extraneous-dependencies */
import { render } from '@testing-library/react'
import CitySelector from '@/components/home/CitySelector'

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    locale: 'fi',
    asPath: '/foo/bar',
    events: {
      on: jest.fn(),
      off: jest.fn()
    },
  })
}))

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (i18nKey) => i18nKey,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe('CitySelector', () => {
  it('should render the CitySelector component', () => {
    const { asFragment } = render(<CitySelector />)
    expect(asFragment()).toMatchSnapshot()
  })
})
