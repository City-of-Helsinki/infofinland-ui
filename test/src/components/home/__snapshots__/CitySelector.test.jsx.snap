// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CitySelector should render the CitySelector component 1`] = `
<DocumentFragment>
  <div
    class="pt-8 border-t border-gray-darker ifu-block--article"
  >
    <div
      class="lg:grid grid-cols-2 pb-8"
    >
      <div
        class="foo"
      >
        <h2
          class="mb-4 text-h2"
        >
          frontpage.cities.title
        </h2>
        <p
          class="mb-8"
        >
          frontpage.cities.text
        </p>
        <button
          aria-haspopup="dialog"
          class="ifu-button"
        >
          frontpage.cities.button
        </button>
      </div>
      <div
        class="flex justify-center"
      >
        <div
          class="hidden lg:block"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; width: initial; height: initial; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; width: initial; height: initial; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27219%27%20height=%27381%27/%3e"
                style="display: block; max-width: 100%; width: initial; height: initial; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt=""
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
